manager/src/main/java/org/openremote/manager/alarm/AlarmService.java

```diff
@@ -352,69 +304,48 @@ private String getAlarmNotificationText(Map<String, String> content) {
     /**
      * Updates an existing alarm if the given user has access to the alarm realm.
      */
-    public void updateAlarm(Long alarmId, String userId, SentAlarm alarm) {
-        SentAlarm oldAlarm = getAlarm(alarmId, userId);
+    public void updateAlarm(SentAlarm oldAlarm, SentAlarm newAlarm) {
         String oldAssigneeId = oldAlarm.getAssigneeId();
-        String newAssigneeId = alarm.getAssigneeId();
-
-        if (newAssigneeId != null) {
-            validateRealmAccessibleToUser(newAssigneeId, alarm.getRealm());
-        }
+        String newAssigneeId = newAlarm.getAssigneeId();
 
         persistenceService.doTransaction(entityManager -> entityManager.createQuery("""
                         update SentAlarm set title=:title, content=:content, severity=:severity, status=:status, lastModified=:lastModified, assigneeId=:assigneeId
                         where id =:id
                         """)
-                .setParameter("id", alarmId)
-                .setParameter("title", alarm.getTitle())
-                .setParameter("content", alarm.getContent())
-                .setParameter("severity", alarm.getSeverity())
-                .setParameter("status", alarm.getStatus())
+                .setParameter("id", oldAlarm.getId())
+                .setParameter("title", newAlarm.getTitle())
+                .setParameter("content", newAlarm.getContent())
+                .setParameter("severity", newAlarm.getSeverity())
+                .setParameter("status", newAlarm.getStatus())
                 .setParameter("lastModified", new Timestamp(timerService.getCurrentTimeMillis()))
                 .setParameter("assigneeId", newAssigneeId)
                 .executeUpdate());
 
-        clientEventService.publishEvent(new AlarmEvent(alarm.getRealm(), PersistenceEvent.Cause.UPDATE));
+        clientEventService.publishEvent(new AlarmEvent(newAlarm.getRealm(), PersistenceEvent.Cause.UPDATE));
 
-        if (alarm.getSeverity() == Alarm.Severity.HIGH) {
-            Set<String> excludeUserIds = Stream.of(userId, oldAssigneeId).filter(Objects::nonNull).collect(Collectors.toSet());
-            sendAssigneeNotification(getAlarm(alarmId, userId), excludeUserIds);
+        if (newAlarm.getSeverity() == Alarm.Severity.HIGH) {
+            Set<String> excludeUserIds = Stream.of(oldAssigneeId).filter(Objects::nonNull).collect(Collectors.toSet());
+            sendAssigneeNotification(oldAlarm, excludeUserIds);
```

Comment:
Shouldn't this be `newAlarm` instead of `oldAlarm` ? Because the assignees may have been updated.